arm-operationalinsights/2015-11-01-preview/swagger/OperationalInsights.json

```diff
@@ -0,0 +1,1230 @@
+            "Succeeded",
+            "Failed",
+            "Canceled",
+            "Deleting",
+            "ProvisioningAccount"
+          ],
+          "x-ms-enum": {
+            "name": "EntityStatus",
+            "modelAsString": true
+          }
+        },
+        "source": {
+          "type": "string",
+          "description": "The source of the workspace.  Source defines where the workspace was created. 'Azure' implies it was created in Azure.  'External' implies it was created via the Operational Insights Portal. This value is set on the service side and read-only on the client side."
+        },
+        "customerId": {
+          "type": "string",
+          "description": "The ID associated with the workspace.  Setting this value at creation time allows the workspace being created to be linked to an existing workspace."
+        },
+        "portalUrl": {
+          "type": "string",
+          "description": "The URL of the Operational Insights portal for this workspace.  This value is set on the service side and read-only on the client side."
+        },
+        "sku": {
+          "$ref": "#/definitions/Sku",
+          "description": "The SKU of the workspace."
+        },
+        "retentionInDays":{
+          "type":"integer",
+          "format": "int32",
+          "minimum": -1,
+          "maximum": 730,
+          "description": "The workspace data retention in days. -1 means Unlimited retention for the Unlimited Sku. 730 days is the maximum allowed for all other Skus. "
+        }
+      },
+      "description": "Workspace properties."
+    },
+    "Workspace": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/WorkspaceProperties",
+          "description":"Workspace properties." 
+        },
+        "eTag": {
+          "type": "string",
+          "description": "The ETag of the workspace."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],    
+      "description": "The top level Workspace resource container."
+    },
+    "WorkspaceListResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Workspace"
+          },
+          "description": "A list of workspaces."
+        }
+      },
+      "description": "The list workspaces operation response."
+    },
+    "Object": {
+      "type": "object",
+      "properties": {},
+      "description": "JSON object"
+    },
+    "Resource": {
+      "properties": {
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource Id"
+        },
+        "name": {
+          "type": "string",
+          "description": "Resource name"
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource type"
+        },
+        "location": {
```

Comment:
@olydis , @fearthecowboy , @amarzavery , @dsgouda , 
Great. I think this is the best solution. Just commited the change to use Proxyresource as a base for Datasource and two other proxy resources. Please take a look. 
Thanks.